src/app/pages/BuyPage/components/BondingCurve/index.tsx

```diff
@@ -0,0 +1,386 @@
+import swapIcon from '../../../../../assets/images/buy/buy_exchange.svg';
+import settingIcon from '../../../../../assets/images/swap/ic_setting.svg';
+import { SlippageDialog } from 'app/pages/BuySovPage/components/BuyForm/Dialogs/SlippageDialog';
+import { useSlippage } from 'app/pages/BuySovPage/components/BuyForm/useSlippage';
+import { weiToNumberFormat } from 'utils/display-text/format';
+import { BuyButton } from 'app/pages/BuySovPage/components/Button/buy';
+import { TxDialog } from 'app/components/Dialogs/TxDialog';
+import { bignumber } from 'mathjs';
+import { Input } from 'app/components/Form/Input';
+import { AvailableBalance } from '../../../../components/AvailableBalance';
+import { Arbitrage } from '../../../../components/Arbitrage/Arbitrage';
+import { useAccount } from '../../../../hooks/useAccount';
+import { getTokenContractName } from '../../../../../utils/blockchain/contract-helpers';
+import { Sovryn } from '../../../../../utils/sovryn';
+import { contractReader } from '../../../../../utils/sovryn/contract-reader';
+import { ErrorBadge } from 'app/components/Form/ErrorBadge';
+import { useMaintenance } from 'app/hooks/useMaintenance';
+import { discordInvite } from 'utils/classifiers';
+import { useSwapsExternal_getSwapExpectedReturn } from '../../../../hooks/swap-network/useSwapsExternal_getSwapExpectedReturn';
+import { useSwapsExternal_approveAndSwapExternal } from '../../../../hooks/swap-network/useSwapsExternal_approveAndSwapExternal';
+import { IPromotionLinkState } from 'app/pages/LandingPage/components/Promotions/components/PromotionCard/types';
+
+import styles from './index.module.scss';
+import { useSwapNetwork_approveAndConvertByPath } from '../../../../hooks/swap-network/useSwapNetwork_approveAndConvertByPath';
+import { useSwapNetwork_conversionPath } from '../../../../hooks/swap-network/useSwapNetwork_conversionPath';
+
+const s = translations.swapTradeForm;
+
+interface Option {
+  key: Asset;
+  label: string;
+}
+
+const xusdExcludes = [Asset.USDT, Asset.DOC];
+
+const tokens = [
+  '0x75bbf7f4d77777730eE35b94881B898113a93124',
+  '0x010C233B4F94d35CaDb71D12D7058aAb58789e8f',
+  '0xaa7038D80521351F243168FefE0352194e3f83C3',
+];
+
+export function BondingCurve() {
+  const { t } = useTranslation();
+  const isConnected = useCanInteract();
+  const { checkMaintenance, States } = useMaintenance();
+  const swapLocked = checkMaintenance(States.SWAP_TRADES);
+
+  const [dialogOpen, setDialogOpen] = useState<boolean>(false);
+  const [amount, setAmount] = useState('');
+  const [sourceToken, setSourceToken] = useState(Asset.RBTC);
+  const [targetToken, setTargetToken] = useState(Asset.SOV);
+  const [sourceOptions, setSourceOptions] = useState<any[]>([]);
+  const [targetOptions, setTargetOptions] = useState<any[]>([]);
+  const [slippage, setSlippage] = useState(0.5);
+  const account = useAccount();
+  const weiAmount = useWeiAmount(amount);
+  // const { value: tokens } = useCacheCallWithValue<string[]>(
```

Comment:
It's okay to remove this part.

Reply:
fixed. but account, weiAmount are requied.